google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java

```diff
@@ -182,9 +176,11 @@ public Blob create(
   @Override
   @Deprecated
   public Blob create(BlobInfo blobInfo, InputStream content, BlobWriteOption... options) {
-    Tuple<BlobInfo, BlobTargetOption[]> targetOptions = BlobTargetOption.convert(blobInfo, options);
-    StorageObject blobPb = codecs.blobInfo().encode(targetOptions.x());
-    Map<StorageRpc.Option, ?> optionsMap = optionMap(targetOptions.x(), targetOptions.y());
+    Opts<ObjectTargetOpt> opts = Opts.unwrap(options).resolveFrom(blobInfo);
+    Map<StorageRpc.Option, ?> optionsMap = opts.getRpcOptions();
+    BlobInfo.Builder builder = blobInfo.toBuilder().setMd5(null).setCrc32c(null);
+    BlobInfo updated = opts.blobInfoMapper().apply(builder).build();
```

Comment:
This is a bit nuanced for me; could you clarify what this apply is doing the BlobInfo builder?

Reply:
Agreed, I was surprised when I ran across the need for it too.

Previously, when converting between certain options (for example[1]) there were actually mutations applied to the BlobInfo.

Instead of doing the same mutation during conversion, we now have a two phase approach. 1) Resolve any extracting Opts against the Info type using `resolveFrom(info)` 2) then applying any mutations to the Info before encoding via the codec.

My hope is that we'll be able to move away from these extractor types of options as they are prone to confusion (Construct/Load a BlobInfo with generation, then specify `generationNotMatch()` resulting in a url like `?generation=3&ifGenerationNotMatch=3` which can't ever succeed). But, for the time being we need them in order to preserve the existing behavior.

[1] https://github.com/googleapis/java-storage/blob/2db5f4a4ebf001248462553a0eb7d4f4399ddfe1/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java#L571-L589